diff --git a/_preload.lua b/_preload.lua
index bee9c24..2b49efe 100644
--- a/_preload.lua
+++ b/_preload.lua
@@ -26,13 +26,13 @@ newaction {
 
 
   onSolution = function(sln)
-    premake.escaper(make.esc)
+    -- premake.escaper(make.esc)
     premake.generate(sln, androidmk.slnApplicationFile(sln), androidmk.generate_applicationmk)
     premake.generate(sln, androidmk.slnAndroidFile(sln), androidmk.generate_androidmk)
   end,
 
   onProject = function(prj)
-    premake.escaper(make.esc)
+    -- premake.escaper(make.esc)
     premake.generate(prj, androidmk.prjFile(prj), androidmk.generate_projectmk)
   end,
 
diff --git a/androidmk.lua b/androidmk.lua
index 513f98d..16b30af 100644
--- a/androidmk.lua
+++ b/androidmk.lua
@@ -44,9 +44,10 @@ androidmk.cflags = {
 }
 
 androidmk.cppflags = {
-  flags = {
+  cppdialect = {
     ["C++11"] = "-std=c++11",
     ["C++14"] = "-std=c++14",
+    ["C++17"] = "-std=c++1z",
   }
 }
 
diff --git a/androidmk_api.lua b/androidmk_api.lua
index 1a95d09..4460d33 100644
--- a/androidmk_api.lua
+++ b/androidmk_api.lua
@@ -68,6 +68,9 @@ api.register {
     "android-18",
     "android-19",
     "android-21",
+    "android-22",
+    "android-23",
+    "android-24",
   },
 }
 
